apps/web/src/store/collateral/hooks.ts

```diff
@@ -434,3 +501,186 @@ export function useCollateralDecimalPlaces() {
+  differenceAmount: BigNumber;
+  xTokenAmount: CurrencyAmount<XToken> | undefined;
+  formattedAmounts: {
+    [x: string]: string;
+  };
+  parsedAmount: {
+    [x: string]: BigNumber;
+    [x: number]: BigNumber;
+  };
+} {
+  const sourceChain = useCollateralXChain();
+  const signedInWallets = useAvailableWallets();
+  const crossChainWallet = useCrossChainWalletBalances();
+  const account = signedInWallets.find(
+    w => xChainMap[w.xChainId].xWalletType === xChainMap[sourceChain].xWalletType,
+  )?.address;
+  const collateralType = useCollateralType();
+
+  const collateralCurrency = React.useMemo(() => {
+    const xToken = xTokenMap[sourceChain].find(t => t.symbol === collateralType);
+    return xToken || SUPPORTED_TOKENS_LIST.find(t => t.symbol === collateralType);
+  }, [collateralType, sourceChain]);
+
+  const availableCollateralAmount: CurrencyAmount<XToken | Currency> | undefined = React.useMemo(() => {
+    return collateralCurrency ? crossChainWallet[sourceChain]?.[collateralCurrency?.wrapped.address] : undefined;
+  }, [collateralCurrency, crossChainWallet, sourceChain]);
+
+  const collateralDecimalPlaces = useCollateralDecimalPlaces();
+  const availableCollateral = useAvailableCollateral();
+  const collateralDeposit = useDepositedCollateral();
+
+  const collateralTotal = useMemo(() => {
+    return availableCollateral.plus(collateralDeposit);
+  }, [availableCollateral, collateralDeposit]);
+
+  const { independentField, typedValue } = useCollateralState();
+  const dependentField: Field = independentField === Field.LEFT ? Field.RIGHT : Field.LEFT;
+
+  //  calculate dependentField value
+  const parsedAmount = React.useMemo(() => {
+    return {
+      [independentField]: new BigNumber(typedValue || '0'),
+      [dependentField]: collateralTotal.minus(new BigNumber(typedValue || '0')),
+    };
+  }, [independentField, dependentField, typedValue, collateralTotal]);
+
+  const formattedAmounts = React.useMemo(() => {
+    return {
+      [independentField]: typedValue,
+      [dependentField]: parsedAmount[dependentField].isZero()
+        ? '0'
+        : parsedAmount[dependentField].toFixed(collateralDecimalPlaces),
+    };
+  }, [independentField, dependentField, typedValue, parsedAmount, collateralDecimalPlaces]);
+
+  const differenceAmount = parsedAmount[Field.LEFT].minus(collateralDeposit);
+
+  const xToken = xTokenMap[sourceChain].find(t => t.symbol === collateralType);
+  const xTokenAmount =
+    xToken && differenceAmount
+      ? CurrencyAmount.fromRawAmount(xToken, differenceAmount.times(10 ** xToken.decimals).toFixed(0))
+      : undefined;
+
+  return {
+    account,
+    sourceChain,
+    collateralType,
+    availableCollateralAmount,
+    collateralDeposit,
+    collateralTotal,
+    formattedAmounts,
+    parsedAmount,
+    collateralDecimalPlaces,
+    differenceAmount,
+    xTokenAmount,
+  };
+}
+
+export function useXCollateralDataByToken(): UseQueryResult<XPositionsRecord[]> {
```

Comment:
rename to useUserPositionData

Reply:
moved to https://github.com/balancednetwork/balanced-network-interface/issues/1388